### PR TITLE
feat(employee): send birthdays reminders to employees in same company

### PR DESCRIPTION
**Pull request by DeeMysterio**
**Handle birthday reminders for multi-company setup:** - 

Send birthday reminders to employees that belong to the same company as that of the birthday guy/girl does. Handle the same for multiple birthdays.

### CODE REVIEW

**Comment by vjFaLk** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/DigiThinkIT/erpnext/pullRequest?prid=4732931&bid=15801796)

File: erpnext/hr/doctype/employee/employee.py [hunk 1]
```diff
@@ -254,15 +254,18 @@ def send_birthday_reminders():
 	if int(frappe.db.get_single_value("HR Settings", "stop_birthday_reminders") or 0):
 		return
 	employees = get_employees_born_today()
-	if employees:
-		recipients_list = frappe.get_all('Employee', filters={'status': 'Active'})
-		recipients = get_employee_emails(recipients_list)
+	companies_with_birthdays = set(employee['company'] for employee in employees)
```
Comment (Alchez) [inline erpnext/hr/doctype/employee/employee.py]: A couple of things:

- Change the variable to something like `companies_with_employee_birthdays`
- Using `dict.get` is safer than simply getting the key

```suggestion
	companies_with_employee_birthdays = set(employee.get('company') for employee in employees)
```

File: erpnext/hr/doctype/employee/employee.py [hunk 2]
```diff
@@ -254,15 +254,18 @@ def send_birthday_reminders():
...
 
+	if employees:
+		recipients_list = frappe.get_all('Employee', fields = ['name', 'company'], filters = {'status': 'Active', 'company': ('in', companies_with_birthdays)})
```
Comment (Alchez) [inline erpnext/hr/doctype/employee/employee.py]: Break this into multiple lines, each for filters and fields, and fix codacy errors on this line

File: erpnext/hr/doctype/employee/employee.py [hunk 3]
```diff
@@ -254,15 +254,18 @@ def send_birthday_reminders():
...
 		birthday_email_template = frappe.db.get_single_value("HR Settings", "birthday_email_template")
 		if birthday_email_template:
 			email_template = frappe.get_doc("Email Template", birthday_email_template)
 
 		for employee in employees:
+			recipients = [{'name': recipient['name']} for recipient in recipients_list if recipient['company'] == employee.company ]
```
Comment (Alchez) [inline erpnext/hr/doctype/employee/employee.py]: Use `dict.get` here as well, and do we need a pass a dictionary to `get_employee_emails`? Wouldn't just the name do?

**Review by Alchez — CHANGES_REQUESTED** [review]

**Review by Alchez — APPROVED** [review]

**Comment by Alchez** [comment]
@DeeMysterio, make this to `bloomstack-v12-staging` as well

**Comment by DeeMysterio** [comment]
> @DeeMysterio, make this to `bloomstack-v12-staging` as well

[here](https://github.com/DigiThinkIT/erpnext/pull/279)

**Pull request merged**